File: 2018_KAKAO_BLIND_RECRUITMENT/1차/다트_게임.py
def solution(dartResult):
    score = 0
    before = 0
    
    index = 0
    for i in range(3):
        if dartResult[index + 1] == '0':
            tmp_score = 10
            index += 2
        else:
            tmp_score = int(dartResult[index])
            index += 1
            
        bonus = dartResult[index]; index += 1
        if bonus == 'D': tmp_score *= tmp_score
        elif bonus == 'T': tmp_score **= 3
        
        if index >= len(dartResult): pass
        elif dartResult[index] == '*':
            score += before
            tmp_score *= 2
            index += 1
        elif dartResult[index] == '#':
            tmp_score = -tmp_score
            index += 1
            
        score += tmp_score
        before = tmp_score
        
    return score

File: 2018_KAKAO_BLIND_RECRUITMENT/1차/다트_게임.cpp
#include <iostream>
using namespace std;

int solution(string dartResult) {
	int score = 0;
	int before = 0;

	int index = 0;
	for (int i = 0; i < 3; i++) {
		// 점수 계산
		int tmp_score;
		if (dartResult[index + 1] == '0') {
			tmp_score = 10;
			index += 2;
		}
		else tmp_score = dartResult[index++] - '0';

		// bonus
		char bonus = dartResult[index++];
		if (bonus == 'D') tmp_score *= tmp_score;
		else if (bonus == 'T') tmp_score = tmp_score * tmp_score * tmp_score;

		// option
		if (index >= dartResult.length());
		else if (dartResult[index] == '*') {
			score += before;
			tmp_score *= 2;
			index++;
		}
		else if (dartResult[index] == '#') {
			tmp_score = -tmp_score;
			index++;
		}

		// 점수 합산
		score += tmp_score;
		before = tmp_score;
	}

	return score;
}
